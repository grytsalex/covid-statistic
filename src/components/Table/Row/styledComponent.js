import styled, { css } from "styled-components";
import { ifProp, prop } from "styled-tools";

export const TableRow = styled.div`
  width: 100%;
  height: 80px;
  border-radius: 20px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.25);
  margin-bottom: 20px;
  display: flex;
  align-items: center;
  cursor: pointer;

  ${ifProp(
    "tableHead",
    css`
      background-color: #2196f3;
      box-shadow: none;
      cursor: auto;
      color: #ffffff;
    `
  )}
`;

export const TableRowItem = styled.div`
  text-transform: capitalize;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;

  &:first-child {
    max-width: 78px;
  }

  &:not(:first-child) {
    border-left: ${prop("borderLeft", "1px solid #B2B2B2")};
    justify-content: left;
    padding-left: 21px;
  }

  &:last-child {
    max-width: 560px;
  }
`;

export const TableText = styled.span`
  font-size: 24px;
  line-height: 28px;
  font-weight: 700;
  cursor: pointer;
`;
