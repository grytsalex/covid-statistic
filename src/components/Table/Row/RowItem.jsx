import React, { memo } from "react";
import { TableRowItem, TableText } from "./styledComponent";
import { PropTypes } from "prop-types";

export const RowItem = memo(({ text, sortData = () => {} }) => {
  return (
    <TableRowItem>
      <TableText onClick={() => sortData(text.toString().replace(/ /g, ""))}>
        {text}
      </TableText>
    </TableRowItem>
  );
});

RowItem.propTypes = {
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  sortData: PropTypes.func,
};
