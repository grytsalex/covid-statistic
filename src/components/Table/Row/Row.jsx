import React, { memo } from "react";
import PropTypes from "prop-types";

import { TableRow } from "./styledComponent";
import { RowItem } from "./RowItem";

export const Row = memo(
  ({
    tableHead = false,
    openModal = () => {},
    sortData = () => {},
    tableData = {},
  }) => {
    return (
      <TableRow tableHead={tableHead} onClick={() => openModal()}>
        {/*<If condition={tableHead}*/}
        {Object.values(tableData).map(({ text, isSorted }, index) => (
          <RowItem
            text={text}
            key={index}
            sortData={isSorted ? sortData : undefined}
          />
        ))}
      </TableRow>
    );
  }
);

Row.propTypes = {
  tableData: PropTypes.object.isRequired,
  tableHeader: PropTypes.bool,
  openModal: PropTypes.func,
  sortData: PropTypes.func,
};
