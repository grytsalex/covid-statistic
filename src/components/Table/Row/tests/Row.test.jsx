import React from "react";
import { Row } from "../Row";
import { shallow } from "enzyme";

describe("Row", () => {
  const commonProps = {
    tableHead: false,
    openModal: jest.fn(),
    sortData: jest.fn(),
    tableData: {},
    withHover: false,
    borderLeft: "",
  };

  it("should match snapshot", () => {
    expect(shallow(<Row {...commonProps} />)).toMatchSnapshot();
  });
});
