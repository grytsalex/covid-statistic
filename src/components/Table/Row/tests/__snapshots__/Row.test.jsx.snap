// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should match snapshot 1`] = `
<styled.div
  onClick={[Function]}
  tableHead={false}
  withHover={false}
/>
`;
